### PR TITLE
feat: new calcite-shell-center-row component and updated slot for calcite-shell

### PR DESCRIPTION
**Pull request by asangma**
There are some things that might need rethinking here, but I am not sure.

One super minor thing is whether we should rename bottom-panel to center-panel and give it an alignment prop for top and bottom.

The major things are

* how do we do `detached` with the bottom slot, right now, I'm just targeting that property
* do we need to make a "center-panel" or "bottom-panel" component in the same way that we have a "shell-panel"?

cc @kat10140 @jcfranco

### CODE REVIEW

**Comment by asangma** [comment]
Generally looking pretty good tho.
image

File: src/components/calcite-shell/calcite-shell.tsx [hunk 1]
```diff
@@ -79,9 +79,11 @@ export class CalciteShell {
       <div class={mainClasses}>
         <slot name={SLOTS.primaryPanel} />
         {this.renderContent()}
-        <slot name={SLOTS.contextualPanel} />
```
Comment (driskull) [inline src/components/calcite-shell/calcite-shell.tsx:82]: @asangma can you revert these changes? We need to keep the `tipManager` until we can do breaking changes. Also, I think contextual panel should be tabbed to before `bottomPanel` right?

Comment (asangma) [reply src/components/calcite-shell/calcite-shell.tsx:82]: re: tipManager
I can revert, but is anybody using that at the moment?

Comment (asangma) [reply src/components/calcite-shell/calcite-shell.tsx:82]: re: contextual panel before centerRow (previously bottomPanel)
If I do that, this happens:
image

I could use `display: grid` on `calcite-shell` > `.main`, but I would like to do that in a different PR maybe?

Comment (asangma) [reply src/components/calcite-shell/calcite-shell.tsx:82]: Hmmm...I could use flex order too I guess...?

Comment (asangma) [reply src/components/calcite-shell/calcite-shell.tsx:82]: Like a dis.
```
slot[name="primary-panel"]::slotted(*) {
  order: 1;
}
slot[name="center-row"]::slotted(*) {
  order: 2;
}
slot[name="contextual-panel"]::slotted(*) {
  order: 3;
}
```
Thoughts?

Comment (asangma) [reply src/components/calcite-shell/calcite-shell.tsx:82]: @driskull Made some updates. Lemme know what you think.

Comment (driskull) [reply src/components/calcite-shell/calcite-shell.tsx:82]: > I can revert, but is anybody using that at the moment?

Lets keep it because if we remove then its a breaking change.

> Hmmm...I could use flex order too I guess...?

We can move the node up if you think the center should come before contextual 👍 

Lets just keep the tipManager slot until we do a larger breaking change

File: src/components/calcite-shell-center-row/calcite-shell-center-row.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+:host([detached]) {
+  animation: calcite-app-fade-in-up var(--calcite-app-animation-time) var(--calcite-app-easing-function);
+  border-radius: var(--calcite-app-border-radius);
+  box-shadow: var(--calcite-app-shadow-0);
+  margin: var(--calcite-app-cap-spacing-half) var(--calcite-app-side-spacing-half)
+    var(--calcite-app-cap-spacing-plus-half);
+  overflow: hidden;
+}
+
+:host([position="end"]) {
```
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.scss:31]: I don't think we need the position do we? Since its always centered?

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  /**
+   * Specifies the maxiumum height of the row.
+   */
+  @Prop({ reflect: true }) heightScale: CalciteScale = "s";
+
+  /**
+   * Arranges the component depending on the elements 'dir' property.
+   */
+  @Prop({ reflect: true }) position: CalcitePosition = "end";
```
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:44]: Do we need this one?

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Hide the content panel.
+   */
+  @Prop({ reflect: true }) collapsed = false;
```
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:24]: Do we need this one?

File: src/components/calcite-shell-center-row/calcite-shell-center-row.scss (hunk 2 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.scss:31]: Ah ha! This is for top and bottom alignment.

`end`
image
 `start`
image

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx (hunk 3 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:44]: See comment with screenshots.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx (hunk 4 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:24]: I assumed we'd want to be somewhat consistent with shell-panel. True or no?

**Comment by asangma** [comment]
Oh! I forgot to add a Storybook.

File: src/components/calcite-shell/calcite-shell.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-shell/calcite-shell.tsx:82]: Sounds good. Thanks man.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.scss (hunk 2 above)
Comment (driskull) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.scss:31]: oh nice! 👍

File: src/components/calcite-shell-center-row/calcite-shell-center-row.tsx (hunk 4 above)
Comment (driskull) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.tsx:24]: We have collapsed because we have the `action-bar` If we `collapse` this panel what is there? @asangma

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import { CSS } from "./resources";
+import { accessible, defaults, hidden, renders } from "../../tests/commonTests";
+
+describe("calcite-shell-center-row", () => {
+  it("renders", async () => renders("calcite-shell-center-row"));
+
+  it("honors hidden attribute", async () => hidden("calcite-shell-panel"));
+
+  it("defaults", async () =>
+    defaults("calcite-shell-panel", [
```
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts:12]: `defaults("calcite-shell-panel", [` should be `calcite-shell-center-row`

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import { newE2EPage } from "@stencil/core/testing";
+
+import { CSS } from "./resources";
+import { accessible, defaults, hidden, renders } from "../../tests/commonTests";
+
+describe("calcite-shell-center-row", () => {
+  it("renders", async () => renders("calcite-shell-center-row"));
+
+  it("honors hidden attribute", async () => hidden("calcite-shell-panel"));
```
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts:9]: `hidden("calcite-shell-panel"))` should be `calcite-shell-center-row`

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts (hunk 5 above)
Comment (driskull) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts:12]: Other property defaults should be added here for testing.

File: src/components/calcite-shell/calcite-shell.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-shell/calcite-shell.tsx:81]: @driskull TipManager slot is back. Nodes reordered. 👍🏽

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts (hunk 5 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts]: I did it! :)

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts (hunk 6 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts]: I did dat now.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts (hunk 5 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts]: Fixed.

**Comment by asangma** [comment]
Thanks @driskull!

**Comment by asangma** [comment]
I shall MURGLE unless you have any concerns, @jcfranco. :)

**Comment by asangma** [comment]
Hurm...there might be a layout bug I overlooked.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    await page.waitForChanges();
+
+    const element = await page.find(`calcite-shell-center-row >>> .${CSS.content}`);
+
+    const isVisible = await element.isVisible();
+
+    expect(isVisible).toBe(true);
+  });
+
+  it("should be accessible", async () => {
```
Comment (jcfranco) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts:41]: FIXME: Drop the curly braces to properly wire up the test. `accessible` returns a promise that needs to be passed as the test (similar to the other tests using the helpers).

```ts
it("should be accessible", async () => 
    accessible("<calcite-shell-center-row><div>content</div></calcite-shell-center-row>")
);
```

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        propertyName: "heightScale",
+        defaultValue: "s"
+      },
+      {
+        propertyName: "position",
+        defaultValue: "end"
+      }
+    ]));
+
+  it("should show row content", async () => {
```
Comment (jcfranco) [inline src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts:27]: What is this test for? It doesn't seem to be testing behavior.

File: src/demos/shell/block-configurations.html [hunk 9]
```diff
@@ -30,7 +30,12 @@
     </style>
 
     <script>
-      require(["esri/WebMap", "esri/views/MapView", "esri/widgets/LayerList","esri/widgets/Zoom"], function(WebMap, MapView, LayerList, Zoom) {
+      require(["esri/WebMap", "esri/views/MapView", "esri/widgets/LayerList", "esri/widgets/Zoom"], function (
```
Comment (jcfranco) [inline src/demos/shell/block-configurations.html:33]: Are these bonus changes? 🎉  Just asking since it doesn't have any row-related changes.

**Comment by asangma** [comment]
Thanks @jcfranco ! I'll make updates.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts (hunk 8 above)
Comment (asangma) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts:27]: Heh...I was totes copying what shell-panel did. I can remove it.

File: src/demos/shell/block-configurations.html (hunk 9 above)
Comment (asangma) [reply src/demos/shell/block-configurations.html:33]: I think that might have happened due to a formatter.

File: src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts (hunk 8 above)
Comment (jcfranco) [reply src/components/calcite-shell-center-row/calcite-shell-center-row.e2e.ts:27]: 👌

File: src/demos/shell/block-configurations.html (hunk 9 above)
Comment (jcfranco) [reply src/demos/shell/block-configurations.html:33]: Gotcha. Thanks for checking. 😄

**Comment by asangma** [comment]
@jcfranco Updated! 🚀

**Comment by driskull** [comment]
Nice @asangma !!! 🥳

**Comment by asangma** [comment]
Thanks you rockers! 🤘🏽

**Comment by jcfranco** [comment]
murgles

MURGLES!
